ml/metamath/ast.py

```diff
@@ -163,6 +163,9 @@ class DisjointStatement(Statement):
     def visit(self, visitor: MetamathVisitor[ResultT]) -> ResultT:
         return visitor.proxy_visit_disjoint_statement(self)  # type: ignore
 
+    def get_metavariables(self) -> set[str]:
+        return {v.name for v in self.metavariables}
+
```

Comment:
While unlikely, these fixes to `get_metavariables` may affect scripts outside of what is intended. Since the CI for github.com/runtimeverification/proof-generation does not have CI for the Haskell backend, could you please double check that this does not break anything @fiedlr ?

Reply:
@nishantjr  Do you mean I should regenerate all the examples with the new proof generation script and verify them?

Reply:
Please, just a small one to make sure things are working.

Reply:
I tried it on `svm` and the proof generation works without issues, the proof verifies even after metamath compresson.